connector/src/main/scala/com/datastax/spark/connector/datasource/CassandraCatalog.scala

```diff
@@ -0,0 +1,431 @@
+package com.datastax.spark.connector.datasource
+
+import java.util
+import java.util.{Locale, Optional}
+
+import com.datastax.oss.driver.api.core.CqlIdentifier
+import com.datastax.oss.driver.api.core.CqlIdentifier.fromInternal
+import com.datastax.oss.driver.api.core.metadata.Metadata
+import com.datastax.oss.driver.api.core.metadata.schema.{ClusteringOrder, TableMetadata}
+import com.datastax.oss.driver.api.querybuilder.SchemaBuilder
+import com.datastax.oss.driver.api.querybuilder.schema._
+import com.datastax.oss.driver.internal.core.metadata.schema.parsing.RelationParser
+import com.datastax.spark.connector.cql.CassandraConnector
+import com.datastax.spark.connector.datasource.CassandraSourceUtil._
+import com.datastax.spark.connector.util.{Logging, NameTools}
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.analysis.{NamespaceAlreadyExistsException, NoSuchNamespaceException, NoSuchTableException, TableAlreadyExistsException}
+import org.apache.spark.sql.connector.catalog.NamespaceChange.{RemoveProperty, SetProperty}
+import org.apache.spark.sql.connector.catalog.TableChange.{AddColumn, DeleteColumn}
+import org.apache.spark.sql.connector.catalog._
+import org.apache.spark.sql.connector.expressions.Transform
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+import org.apache.spark.{SparkConf, SparkEnv}
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+class CassandraCatalogException(msg: String) extends IllegalArgumentException(msg)
+
+/**
+  * A Spark Sql Catalog for inter-operation with Cassandra
+  *
+  * Namespaces naturally map to C* Keyspaces, but they are always only a single
+  * element deep.
+  */
+class CassandraCatalog extends CatalogPlugin
```

Comment:
We do not, here is my doc on it for clarity (will be in the next commit)

> DatasourceV2 makes connecting to Cassandra now easier than ever. Parameters for configuring your connection
> can be done in the SparkConf, SparkSession, spark-defaults file or individually for the Catalog.  Once a catalog is configured it can 
> be accessed through both SparkSql and DataFrames to read from, write to, create, and drop Cassandra tables.
> 
> While setting up a Catalog can be done before starting your application it can also be done at 
> runtime by setting  ```spark.sql.catalog.anyname``` to  ```com.datastax.spark.connector.datasource.CassandraCatalog```
> In your SparkSession. The "anyname" portion can be any name you would like to specify for this 
> particular catalog.
> 
> Then any parameter for that catalog can be set by just appending the parameter name to the catalog name as in
> ```spark.sql.catalog.anyname.propertykey``` to ```propertyvalue```
> 
> Example: Manually configuring a Catalog named Cass100
> 
> ```scala
> spark.conf.set(s"spark.sql.catalog.cass100", "com.datastax.spark.connector.datasource.CassandraCatalog")
> spark.conf.set(s"spark.sql.catalog.cass100.spark.cassandra.connection.host", "127.0.0.100")
> ```
> 
> Once this is set, we can access the catalog by using the triple part identifier of 
> ```catalog.keyspace.table``` which in this case would be ```cass100.ks.tab```